src/transactions/saga.ts

```diff
@@ -225,7 +233,62 @@ function* watchAddressToE164PhoneNumberUpdate() {
   )
 }
 
+export function* getTransactionReceipt(
+  transaction: StandbyTransaction & { transactionHash: string },
+  network: Network
+) {
+  const { transactionHash } = transaction
+
+  try {
+    const receipt = yield* call(publicClient[network].getTransactionReceipt, {
+      hash: transactionHash as Hash,
+    })
+
+    if (receipt) {
+      yield* put(
+        transactionConfirmed(transaction.context.id, {
+          transactionHash: receipt.transactionHash,
+          block: receipt.blockNumber.toString(),
+          status: receipt.status == 'success',
+        })
+      )
+    }
+  } catch (e) {
+    Logger.warn(
+      TAG,
+      `Error found when trying to fetch status for transaction with hash: ${transactionHash} in ${network}`,
+      (e as Error).message
+    )
+  }
+}
+
+export function* internalWatchPendingTransactionsInNetwork(network: Network) {
+  const pendingStandbyTransactions = yield* select(pendingStandbyTransactionsSelector)
+  const filteredPendingTxs = pendingStandbyTransactions.filter((tx) => {
+    return tx.networkId === networkConfig.networkToNetworkId[network] && tx.transactionHash
+  })
+
+  for (const transaction of filteredPendingTxs) {
+    yield* fork(getTransactionReceipt, transaction, network)
+  }
+}
+
+function* watchPendingTransactionsInNetwork(network: Network) {
+  while (true) {
+    yield* call(internalWatchPendingTransactionsInNetwork, network)
+    yield* delay(WATCHING_DELAY_BY_NETWORK[network])
+  }
+}
+
+function* watchPendingTransactions() {
+  const supportedNetworks = Object.keys(publicClient) as Network[]
```

Comment:
i started thinking about how this affects most users once shipped, we will be spawning a new saga every few seconds that checks for ethereum pending tx's when they're guaranteed to have none. the impact is probably small, but it seems like something we can already optimise for by collecting the enabled networks in the multichain dynamic config? since pending tx's are only enabled for send and swaps, we could already know which networks we need to use

Reply:
yes! except we want the union, not the intersection :D

Reply:
The union between send flow + swap flow, but the intersection with viem.
We can't process networks that viem doesn't support